### PR TITLE
Integration testing

### PR DESCRIPTION
**Pull request by ppmarq1**
# In this milestone we

- Used Capybara to write integration tests for each view in the project.

- [ ] For the user index page:

        _can see the username of all other users._
        _I can see the profile picture for each user._
        _I can see each user's number of posts._
       _When I click on a user, I am redirected to that user's show page._

- [ ] For the user show page:

        _I can see the user's profile picture._
        _I can see the user's username._
        _I can see the number of posts the user has written._
        _I can see the user's bio._
        _I can see the user's first 3 posts._
        _I can see a button that lets me view all of a user's posts._
        _When I click a user's post, it redirects me to that post's show page._
        _When I click to see all posts, it redirects me to the user's post's index page._

- [ ] For the user post index page:

         _I can see the user's profile picture._
         _I can see the user's username._
         _I can see the number of posts the user has written._
         _I can see a post's title._
         _I can see some of the post's body._
         _I can see the first comments on a post._
         _I can see how many comments a post has._
         _I can see how many likes a post has._
         _I can see a section for pagination if there are more posts than fit on the view._
         _When I click on a post, it redirects me to that post's show page._

- [ ] For the user post show page: 

          _I can see the post's title.
          _I can see who wrote the post._
          _I can see how many comments it has._
          _I can see how many likes it has._
          _I can see the post body._
          _I can see the username of each commentor._
          _I can see the comment each commentor left._

### CODE REVIEW

File: app/views/layouts/application.html.erb [hunk 1]
```diff
@@ -5,12 +5,15 @@
     <meta name="viewport" content="width=device-width,initial-scale=1">
     <%= csrf_meta_tags %>
     <%= csp_meta_tag %>
+    <%= stylesheet_link_tag "tailwind", "inter-font", "data-turbo-track": "reload" %>
```
Comment (danmainah) [inline app/views/layouts/application.html.erb]: - [x] Kindly check how you setup tailwind, am getting this error whenever I run your tests:
image
`This leads to failure of your tests`
image

**Review by danmainah — CHANGES_REQUESTED** [review]
Hi @ppmarq1 and @stanleySimeon , 

Good job so far!
There are some issues that you still need to work on to go to the next project but you are almost there!

## Highlights
✔️ No linter errors
✔️ Github flow was correctly used


# Required Changes ♻️

- [x] Kindly solve the N+1 problem in your action creator didn't see any solution.

_Check the comments under the review._

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._


Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s).

File: app/views/layouts/application.html.erb (hunk 1 above)
Comment (stanleySimeon) [reply app/views/layouts/application.html.erb]: Hello @ppmarq1, I didn't get it on my side

Comment (stanleySimeon) [reply app/views/layouts/application.html.erb]: Kindly run `bundle exec rake assets:precompile` to solve it in your side
Thanks

**Review by Davidosky007 — COMMENTED** [review]
Hi @ppmarq1 👋, 

Great work on making the changes requested by a previous reviewer 👏 
You've done well implementing some of the requested changes, but there are still some which aren't addressed yet.


 ## Highlights
* No linter errors 👍 
* Well PR description 👍 
* Proper use of Gitflow 👍 

 ## Required Changes ♻️
* the error still persit even after running ` bundle exec rake assets:precompile` 
image

_Check the comments under the review._

## Optional suggestions

_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application._


Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me @davidosky007 in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**


------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Comment by stanleySimeon** [comment]
Hello @Davidosky007, Everything work perfectly on my side
How do you think about it?

**Comment by Davidosky007** [comment]
> Hello @Davidosky007, Everything work perfectly on my side
> 
> How do you think about it?
> 
> 

Kindly update your readme file with the steps you took and also add a screenshot of your results so that the next reviewer will approve it

**Pull request merged**